### PR TITLE
scanner/yara: Improve compiler diagnostics

### PR DESCRIPTION
**Pull request by hillu**
This improves a bit on #38

### CODE REVIEW

File: scanner/yara/compile.go [hunk 1]
```diff
@@ -110,13 +110,25 @@ func compile(what int, inputfiles []string) (*yr.Rules, error) {
...
 	}
+	whats := [...]string{"file", "process"}[what]
 	rs, err := c.GetRules()
-	if c.Warnings != nil && config.YaraFailOnWarnings {
-		w := c.Warnings[0]
-		return nil, fmt.Errorf("Yara emits at least one warning. %s:%d %s", w.Filename, w.Line, w.Text)
-	}
 	if err != nil {
-		return nil, err
+		for _, e := range c.Warnings {
```
Comment (0mbi) [inline scanner/yara/compile.go:116]: Is this a typo: c.Warnings -> c.Errors?

File: scanner/yara/compile.go [hunk 2]
```diff
@@ -110,13 +110,25 @@ func compile(what int, inputfiles []string) (*yr.Rules, error) {
 			return nil, err
 		}
 	}
+	whats := [...]string{"file", "process"}[what]
```
Comment (0mbi) [inline scanner/yara/compile.go:113]: Name this maybe file_or_process? _what_ is already kind of too generic for naming, but _whats_ won't make it clearer here.

File: scanner/yara/compile.go (hunk 1 above)
Comment (hillu) [reply scanner/yara/compile.go:116]: right.

File: scanner/yara/compile.go (hunk 2 above)
Comment (hillu) [reply scanner/yara/compile.go]: Using "what" was a bad idea, too, renamed both.

**Comment by 0mbi** [comment]
Ok, great. Can be merged imho!

**Pull request merged**